crowbar_framework/lib/crowbar/validator/network_validator.rb

```diff
@@ -4,7 +4,7 @@ class NetworkValidator
       def validate_network
         `#{Rails.root.join("..", "bin/network-json-validator").to_s} \
           --admin-ip \
-          #{Socket.ip_address_list.detect(&:ipv4_private?).ip_address} \
+          #{IPSocket.getaddress(Socket.gethostname)} \
```

Comment:
> Socket.gethostname returns the FQDN from /etc/hostname

Sorry for nitpicking here, but technically gethostname contains what ever was configured as the hostname (using hostnamectl or hostname <hostname>). It doesn't read /etc/hostname. And in many cases this is just the unqualified hostname  (without the domain).
IPSocket.getaddress will then use the resolver (DNS and/or /etc/hosts or whatever is configured in nsswitch) to figure out the FQDN and the corresponding IP address.</nitpick>